src/mappings/mappingHandlers.ts

```diff
@@ -0,0 +1,267 @@
+    } 
+    let accountid = poolrecord.owner.toString();
+    let int_ownerreward = (ownerreward as Balance).toNumber();
+    let int_stakerinterest = (stakerinterest as Balance).toNumber();
+    
+    let str_accountid = accountid.toString();
+    let block_key = blake2AsHex(String(blockid) + ' ' + String(pid) + ' ' + accountid);
+    let block_owner_record = await AccountOwnerRewardInBlock.get(block_key);
+    if (block_owner_record == undefined) {
+        let block_owner_record1 = new AccountOwnerRewardInBlock(block_key);
+        block_owner_record1.blockid = BigInt(blockid.toString());
+        block_owner_record1.pid = BigInt(pid.toString());
+        block_owner_record1.accountid = accountid.toString();
+        block_owner_record1.balance = int_ownerreward;
+        await block_owner_record1.save();
+    } else {
+        block_owner_record.balance += int_ownerreward;
+        await block_owner_record.save();
+    }
+    
+    let date = new Date();
+    let date_key = blake2AsHex(date.toLocaleDateString + ' ' + String(pid) + ' ' + accountid);
+    let date_owner_record = await AccountOwnerRewardInDay.get(date_key);
+    if (date_owner_record == undefined) {
+        let date_owner_record1 = new AccountOwnerRewardInDay(date_key);
+        date_owner_record1.day = date.toLocaleDateString();
+        date_owner_record1.pid = BigInt(pid.toString());
+        date_owner_record1.accountid = accountid.toString();
+        date_owner_record1.balance = int_ownerreward;
+        await date_owner_record1.save();
+    } else {
+        date_owner_record.balance += int_ownerreward;
+        await date_owner_record.save();
+    }
+    
+    
+
+    if (poolrecord.shares > 0) {
+        for (var idx = 0; idx < poolrecord.poolstakers.length; idx++) {
+            let accountid = poolrecord.poolstakers[idx];
+            let user_key = blake2AsHex(String(pid) + ' ' + accountid);
+            let record = await PoolStakersShares.get(user_key);
+            if (record == undefined) {
+                let blockid = event.block.timestamp.toString();
+                let errorkey = blake2AsHex(blockid + ' ' + ErrorType.StakerNotFound + ' ' + str_pid + ' ' + str_accountid);
+                let error_record = new ErrorRecord(errorkey);
+                error_record.errortype = ErrorType.StakerNotFound;
+                let error_map = new Map();
+                error_map["pid"] = str_pid;
+                error_map["accountid"] = str_accountid;
+                error_record.blockid = blockid;
+                error_record.error = JSON.stringify(error_map);
+                await error_record.save();
+                return
+            }
+            let block_key = blake2AsHex(String(blockid) + String(pid) + ' ' + accountid);
+            let date_key = blake2AsHex(date.toLocaleDateString + String(pid) + ' ' + accountid);
+            let staker_interest = int_stakerinterest * parseFloat(record.shares.toString()) / parseFloat(poolrecord.shares.toString());
```

Comment:
Should we be aware of the return value of `parseFloat`?. e.g. make sure it's not a NaN